deephaven/vscode-deephaven
# feat: Groovy support

This PR adds support for Groovy servers. It also adds smarter logic to hide / show the connection status bar item based on context.

- If no servers are connected, only show the status bar item if the active file has a languageid for a configured server (python or groovy)
- If server is connected, show the status bar item regardless of active file or config. This allows user to explicitly disconnect from server
- Also fixed #74 where script was being re-opened if another tab group is active

### Testing
Create a python + groovy file in a vscode workspace

```python
# test.py
print("testing")
```
```groovy
// test.groovy
table = emptyTable(10)
```

- Default extension config should have the single python config for http://localhost:10000. You should only see the connection status bar item if you are in a python file, or if you have already connected to the server
- If you have already connected to the default python server and run a groovy file, should see error message "No Deephaven server configured for console type: 'groovy'"
- Config for a python + groovy server
  ```json
  {
    "vscode-deephaven.core-servers": [
      "http://localhost:10000/",
      {
        "consoleType": "groovy",
        "url": "http://localhost:10001/"
      }
    ]
  }
  ```
  Should only show status bar item if python or groovy file is selected or if already connected. Assuming the servers are actually running, running either file should automatically run it against the correct server.
  
For the issue of the extension re-opening the script being run in another tab group:
- Run a python file that produces a panel
- Click on the output panel (this changes the active tab group)
- Run the python file again without selecting the tab first
- File should no longer be reopened in the tab group of the active panel

resolves #63
resolves #74
resolves #34
